src/tests/ftest/container/fill_destroy_loop.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2019-2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+import random
+
+from apricot import TestWithServers
+from general_utils import get_random_bytes, DaosTestError
+from test_utils_container import TestContainerData
+
+
+class BoundaryPoolContainerSpace(TestWithServers):
+    """Test class for pool Boundary testing.
+
+    Test Class Description:
+        Test to create a pool and container and write random data to fill the container and delete
+        container, repeat the test 100 times as boundary condition.
+
+    :avocado: recursive
+    """
+
+    DER_NOSPACE = "-1007"
+
+    def write_pool_until_nospace(self):
+        """write pool and container until pool is full. """
+        cont_obj_class = self.params.get("obj_class", "/run/container/*")
+        block_size = self.params.get("block_size", "/run/container/*")
+        akey_size = self.params.get("akey_size", "/run/container/*")
+        dkey_size = self.params.get("dkey_size", "/run/container/*")
+        # select a random data size based on block_size
+        data_size = random.randint(block_size*0.5, block_size*1.5) # nosec
+
+        container = self.get_container(self.pool)
+        container.open()
+        written_byte = 0
+        write_count = 1
+        data_str = get_random_bytes(data_size)
+        while True:
+            dkey = get_random_bytes(dkey_size)
+            akey = get_random_bytes(akey_size)
+            try:
+                written_byte += data_size
+                self.d_log.debug("--{0}writing cont-obj {1} byte data, total {2}..".format(
+                    write_count, data_size, written_byte))
+                container.written_data.append(TestContainerData(False))
+                container.written_data[-1].write_record(
+                    container, akey, dkey, data_str, obj_class=cont_obj_class)
+                self.d_log.debug("--{0}wrote cont-obj, sz {1}".format(write_count, data_size))
+                write_count += 1
+            except DaosTestError as excep:
+                if self.DER_NOSPACE in repr(excep):
```

Comment:
(pylint-no-else-break) Unnecessary "else" after "break", remove the "else" and de-indent the code inside it